components/TransactionForm.js

```diff
@@ -0,0 +1,106 @@
+import { useRouter } from "next/router";
+import styled from "styled-components";
+
+export default function TransactionForm({ onAddEntry, title }) {
+  function handleSubmit(event) {
+    event.preventDefault();
+
+    const formData = new FormData(event.target);
+    const data = Object.fromEntries(formData);
+
+    onAddEntry(data);
+  }
+
+  const router = useRouter();
+
+  return (
+    <main>
+      <StyledHeader>
+        <StyledHeadline>{`${title}`}</StyledHeadline>
+      </StyledHeader>
+      <StyledForm onSubmit={handleSubmit}>
+        <StyledLabel htmlFor="spending">ausgegeben:</StyledLabel>
+        <StyledInput
+          type="number"
+          step="0.01"
+          min="0"
+          id="spending"
+          name="spending"
+          placeholder="Betrag in €"
+          required
+        />
+        <StyledLabel htmlFor="spendFor">für:</StyledLabel>
+        <StyledInput
+          type="text"
+          id="spendFor"
+          name="spendFor"
+          placeholder=" kurze Beschreibung"
+          required
+        />
+        <ButtonGroup>
+          <StyledCancelButton onClick={() => router.push("./overview")}>zurück</StyledCancelButton>
+          <StyledSubmitButton type="submit" onClick={() => router.push("./overview")}>
+            bestätigen
+          </StyledSubmitButton>
+        </ButtonGroup>
+      </StyledForm>
+    </main>
+  );
```

Comment:
There are a few changes in this component that I would like to recommend to you.

- `<StyledCancelButton>` is actually a Link and should not be rendered inside the `<form>`
- `<StyledSubmitButton>` the `onClick `could lead to mean bugs, because the button is `type="submit"`. Better call `router.push("./overview)` inside the `handleSubmit` function redirect to `page/overview`
- You should delete the `<ButtonGroup>` component and find an other solution for the styling - sorry 🫣



```suggestion
  return (
    <main>
      <StyledHeader>
        <StyledHeadline>{`${title}`}</StyledHeadline>
      </StyledHeader>
      <StyledForm onSubmit={handleSubmit}>
        <StyledLabel htmlFor="spending">ausgegeben:</StyledLabel>
        <StyledInput
          type="number"
          step="0.01"
          min="0"
          id="spending"
          name="spending"
          placeholder="Betrag in €"
          required
        />
        <StyledLabel htmlFor="spendFor">für:</StyledLabel>
        <StyledInput
          type="text"
          id="spendFor"
          name="spendFor"
          placeholder=" kurze Beschreibung"
          required
        />
      <StyledSubmitButton type="submit">bestätigen</StyledSubmitButton>
      </StyledForm>
      <GoBackLink href='/overview'>zurück</GoBackLink>
    </main>
  );
```